src/c++/perf_analyzer/infer_data_manager.cc

```diff
@@ -99,7 +99,11 @@ InferDataManager::CreateAndPopulateInput(
     if (data_ptrs[i] == nullptr) {
       missing_data_cnt++;
     } else {
-      RETURN_IF_ERROR(input->AppendRaw(data_ptrs[i], byte_size[i]));
+      // FIXME TKG -- Another possible fix here would have been to update
+      // AppendRaw to handle empty byte size case
+      if (byte_size[i]) {
```

Comment:
Another case that came up was that InferInput wasn't happy if you call AppendRaw with byte_size of 0. It would still [add the empty pointer to a list](https://github.com/triton-inference-server/client/blob/main/src/c%2B%2B/library/common.cc#L149) which seemed to screw up the InferInput's internal iterator. A better fix might be to not add the pointer to the list for input_byte_size==0, but I wanted to talk to @Tabrizian first.

Reply:
This looks like a bug in the client libraries.

Reply:
ok, I'll put up a proposed fix today. I just don't have enough visibility into the clients to know if the fix would break anything else.

Reply:
I have reverted this workaround. Instead of moving the workaround to InferInput, I dug deeper and I believe I root caused the hang to an infinite loop in the http client.  grpc client doesn't seem to have the same issue.

@Tabrizian and @jbkyang-nvi  please review the change to http_client.cc